media/json-binding/src/main/java/org/glassfish/jersey/jsonb/internal/JsonBindingProvider.java

```diff
@@ -70,6 +74,9 @@ public Object readFrom(Class<Object> type, Type genericType,
                            MediaType mediaType,
                            MultivaluedMap<String, String> httpHeaders,
                            InputStream entityStream) throws IOException, WebApplicationException {
+        if (entityStream.available() == 0) {
```

Comment:
Fixed in 2.30 in several commits, last is 1c3b8d5e031814da04c5a0050ca1e6a1c0faf52c

Reply:
entityStream.available() can return 0 also if came only headers and body is not buffered yet.
isEmpty would be correct.